### DISCUSSION TITLE
docs/g4install/chap2_guide/ubuntu/g4install-111-ubt2204/

### DISCUSSION THREAD

**Discussion by giscus** [evandde/blog_comment#36] [Announcements]
# docs/g4install/chap2_guide/ubuntu/g4install-111-ubt2204/

설치 상세: Ubuntu 22.04 LTS, Geant4-11.1.0, Qt5 연동, Multi-threading 활성화

https://evanote.mew.kr/docs/g4install/chap2_guide/ubuntu/g4install-111-ubt2204/

**Comment by yoon-Seo-98** [comment]
안녕하세요.
ubuntu 에서 sudo make -j && sudo make install 실행할 경우 아래 error가 뜹니다.. 뭐가 잘못되었을까요 ..?



CMake Error at /opt/geant4-v11.1.0/build/Externals/G4NDL-4.7/src/G4NDL-stamp/extract-G4NDL.cmake:51 (file):
  file RENAME failed to rename

    /opt/geant4-v11.1.0/build/data/ex-G4NDL1234/G4NDL4.7

  to

    /opt/geant4-v11.1.0/build/data/G4NDL4.7

  because: Permission denied



make[2]: *** [CMakeFiles/G4NDL.dir/build.make:100: Externals/G4NDL-4.7/src/G4NDL-stamp/G4NDL-download] Error 1
make[1]: *** [CMakeFiles/Makefile2:925: CMakeFiles/G4NDL.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

**Reply by evandde** [reply]
안녕하세요~

일단 Permission denied error가 뜬 것으로 봤을 때, `sudo` 권한이 없는 계정이 아닌가 의심이 됩니다만....
그 이전에 `cmake` 과정까지는 문제가 없다가 저기서만 에러가 나는 것도 이상하네요

데이터파일을 다운받은 뒤 압축 풀고 경로명을 변경하는 과정에서 권한오류가 뜬 것으로 보아, 다운로드 과정에서 뭔가 충돌이 났거나 하는 문제도 의심해볼 수 있을 것 같습니다.

일단은 build 폴더 자체를 완전히 삭제하신 뒤,
`sudo cmake` 수행 -> `sudo make` 수행 -> `sudo make install` 수행
을 순차적으로 다시 해보시는 것을 권유드립니다.

문제가 지속된다면, 데이터파일을 직접 다운로드 받은 뒤에 경로를 따로 지정해주는 방법을 택해야 할 수도 있을 것 같네요

**Reply by yoon-Seo-98** [reply]
감사합니다 ! 


make와 make install까지 진행 후 
env | grep Geant4를 했을 때 
GEANT4_DATA_DIR=/opt/geant4-v11.1.0/install/share/Geant4/data
이것만 뜨고 PATH가 뜨지 않는 이유를 알 수 있을까요 ? ㅠㅠ

**Reply by jhb5648** [reply]
저도 똑같이 PATH가 뜨지 않았지만, 정상 실행되는 것 확인했습니다.. 참고하세요!

**Comment by yoon-Seo-98** [comment]
안녕하세요 !

./exampleB1을 했을 때

Got keys from plugin meta data ("vnc")
QFactoryLoader::QFactoryLoader() looking at "/usr/lib/x86_64-linux-gnu/qt5/plugins/platforms/libqxcb.so"
Found metadata in lib /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms/libqxcb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "xcb"
        ]
    },
    "archreq": 0,
    "className": "QXcbIntegrationPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("xcb")
QFactoryLoader::QFactoryLoader() checking directory path "/home/yoonseo/B1/build/platforms" ...
loaded library "/usr/lib/x86_64-linux-gnu/qt5/plugins/platforms/libqxcb.so"
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

Aborted (core dumped)

이러한 오류 코드가 뜹니다 ...
구글에 아무리 찾아봐도 해결 방법이 없는데 혹시 해결 할 수 있는 방법이 있을까요 ?

안된다면 Qt5 OpenGL을 다시 삭제하고 설치하고 싶은데 혹시 삭제하고 설치하는 방법도 알 수 있을까요 ?

**Reply by evandde** [reply]
혹시 실제 Ubuntu 운영체제 자체를 사용하고 계시지 않고, Windows 내에 WSL의 형태로 Ubuntu 환경을 만들어 설치하셨나요?
만약 그렇다면 [이 글](https://evanote.mew.kr/post/201028-cannot-open-libqt5core/)부터 참고해보시기 바랍니다.

명령줄에 `qmake`를 입력하였을 때 제대로 실행되어 사용법 매뉴얼이 출력되는지,
Geant4 설치하실 때 `cmake` 단계에서 Qt 연동을 하셨는지 검토해보실 필요도 있습니다.

Qt를 삭제하고 재설치하려면, `sudo apt remove qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools`와 같이 삭제하시면 됩니다.
설치할 때 `install`이라고 썼던 부분을 `remove`로 바꾸시면 됩니다.

**Comment by jjunsung** [comment]
안녕하세요~ 도움 많이 받고 있습니다!!
저도 PATH 환경설정에서 문제를 해결하지 못해 댓글 남겨봅니다!!

echo "source /opt/geant4-v11.1.0/install/bin/geant4.sh" >> ~/.bashrc

를 통해 환경변수 설정해주었지만 반영되지 않네요... 수동으로 작성해보려 해도 쉘스크립트 파일이라 [Read-Only]라 수정이 불가하네요

경험 있으시다면 도움 부탁 드립니다. 감사합니다.

**Reply by evandde** [reply]
핵심은 `~` 디렉토리 내에 있는 `.bashrc`라는 문서 안에, `source /opt/geant4-v11.1.0/install/bin/geant4.sh` 라는 문장이 적혀있어야 한다는 것입니다.

리눅스 상에서 gedit든, vim이든, nano든, @jjunsung님이 사용하기 편한 텍스트에디터로 `.bashrc` 파일을 열고, `source /opt/geant4-v11.1.0/install/bin/geant4.sh` 문장을 적어주시기 바랍니다.

만약, `.bashrc` 파일이 Read-only라서 수정이 불가능하다면 이 때는 권한 문제를 살펴보셔야 할 것 같습니다.

원래 계정의 소유자라면, 본인 계정의 home 디렉토리(`~`) 내에 있는 `.bashrc` 파일을 자유롭게 수정할 수 있어야 합니다.

따라서, 이 부분은 `ls -al ~/.bashrc`를 통해 `.bashrc` 파일의 소유자와 권한을 살펴보시고, 현재 접속한 계정으로 해당 파일의 수정권한이 있는 것이 맞는지 확인해보시기 바랍니다.

**Comment by jhjrdna** [comment]
안녕하세요! 
게시글을 읽고 ubuntu에 설치는 잘 했습니다! 감사합니다!
레드햇 기반 centos 에 설치하려고 하는데 위에 설명해주신 pre-requisite 과 환경변수 설정들의 내용은 모두 같고 명령어만 변경해서 설치하면 될까요?

**Reply by evandde** [reply]
넵 centos에서는 apt대신 yum을 쓰시면되는데, 라이브러리들을 다운받기위한 이름이 아마 apt랑 조금 다를수도있으니 확인이 필요합니다.

예를 들면 `apt install build-essential` 대신 `yum groupinstall "development tools"` 를 써야하는 등의 차이가 있습니다.

**Comment by Fluence0115** [comment]
선생님 감사합니다! 정말 많은 도움이 되었습니다.

**Comment by parksungkuk1** [comment]
진심으로 정말 감사합니다. 

ubuntu를 활용하여 GEANT4 설치를 완료하고 제시해 주신 예제까지 실행하여 GUI 창까지 잘 띄웠습니다.

그러나 설명 캡쳐상에 보이는 구조물은 보이지 않고 까만 바탕만 띄워주는 원인이 있을까요?

**Reply by evandde** [reply]
GUI까지 떴는데 구조물이 보이지 않는다면 vis.mac이 제대로 호출되지 않는 등의 문제일 수 있습니다.

구체적인 에러문구 등을 알려주셔야 판단이 가능할 것 같습니다.
